include/at_cmd_parser/at_params.h

```diff
@@ -43,6 +43,12 @@ enum at_param_type {
 	AT_PARAM_TYPE_ARRAY,
 	/** Empty or optional parameter that should be skipped. */
 	AT_PARAM_TYPE_EMPTY,
+	/** Parameter is a SET AT command. */
+	AT_PARAM_TYPE_CMD_SET,
```

Comment:
I think it is a bit strange that the at params module should know anything about what command type. I think testing for which command type it is should be a separate helper function.

`
if (at_params_parse_cmd_type(at_params_get_cmd(list)) == AT_CMD_TYPE_TEST) {
        /* Some code */
}
`

Reply:
I've just updated the PR, new version more or less matches what you say, please check.